test.js

```diff
@@ -19,325 +19,341 @@
+      var a1 = [1, 2, 3];
+      var a2 = [1, 2, 3];
+      a1.a = 'test';
+      a1.b = true;
+      a2.b = true;
+      a2.a = 'test';
+      assert.throws(makeBlock(assert.deepEqual, keys(a1), keys(a2)),
+                assert.AssertionError);
+      assert.doesNotThrow(makeBlock(assert.deepEqual, a1, a2));
+    });
 
-test('assert - Testing the throwing', function () {
-  var aethrow = makeBlock(thrower, assert.AssertionError);
-  aethrow = makeBlock(thrower, assert.AssertionError);
-
-  // the basic calls work
-  assert.throws(makeBlock(thrower, assert.AssertionError),
-                assert.AssertionError, 'message');
-  assert.throws(makeBlock(thrower, assert.AssertionError), assert.AssertionError);
-  assert.throws(makeBlock(thrower, assert.AssertionError));
-
-  // if not passing an error, catch all.
-  assert.throws(makeBlock(thrower, TypeError));
-
-  // when passing a type, only catch errors of the appropriate type
-  var threw = false;
-  try {
-    assert.throws(makeBlock(thrower, TypeError), assert.AssertionError);
-  } catch (e) {
-    threw = true;
-    assert.ok(e instanceof TypeError, 'type');
-  }
-  assert.equal(true, threw,
-               'a.throws with an explicit error is eating extra errors',
-               assert.AssertionError);
-  threw = false;
-
-  // doesNotThrow should pass through all errors
-  try {
-    assert.doesNotThrow(makeBlock(thrower, TypeError), assert.AssertionError);
-  } catch (e) {
-    threw = true;
-    assert.ok(e instanceof TypeError);
-  }
-  assert.equal(true, threw,
-               'a.doesNotThrow with an explicit error is eating extra errors');
-
-  // key difference is that throwing our correct error makes an assertion error
-  try {
-    assert.doesNotThrow(makeBlock(thrower, TypeError), TypeError);
-  } catch (e) {
-    threw = true;
-    assert.ok(e instanceof assert.AssertionError);
-  }
-  assert.equal(true, threw,
-               'a.doesNotThrow is not catching type matching errors');
-});
-
-test('assert.ifError', function () {
-  assert.throws(function() {assert.ifError(new Error('test error'))});
-  assert.doesNotThrow(function() {assert.ifError(null)});
-  assert.doesNotThrow(function() {assert.ifError()});
-});
-
-test('assert - make sure that validating using constructor really works', function () {
-  var threw = false;
-  try {
-    assert.throws(
-        function() {
-          throw ({});
-        },
-        Array
-    );
-  } catch (e) {
-    threw = true;
-  }
-  assert.ok(threw, 'wrong constructor validation');
-});
-
-test('assert -  use a RegExp to validate error message', function () {
-  assert.throws(makeBlock(thrower, TypeError), /test/);
-});
-
-test('assert - se a fn to validate error object', function () {
-  assert.throws(makeBlock(thrower, TypeError), function(err) {
-    if ((err instanceof TypeError) && /test/.test(err)) {
-      return true;
-    }
-  });
-});
-
-test('assert - Make sure deepEqual doesn\'t loop forever on circular refs', function () {
```

Comment:
this test I modified because instead of throwing an error it was just terminating early with the correct answer, seemed like an even more desirable property